### PR TITLE
[TEVA-3038] Display job roles on review and job listing page

### PR DESCRIPTION
**Pull request by cesidio**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-3038

## Screenshots of UI changes:
Screenshot 2021-08-31 at 16 28 41
Screenshot 2021-08-31 at 16 29 08
Screenshot 2021-09-01 at 11 34 52

### CODE REVIEW

File: app/presenters/vacancy_presenter.rb [hunk 1]
```diff
@@ -75,10 +75,16 @@ def show_primary_job_role
     I18n.t("helpers.label.publishers_job_listing_job_role_form.primary_job_role_options.#{primary_job_role}")
   end
 
+  def additional_job_role(role)
+    I18n.t("helpers.label.publishers_job_listing_job_role_details_form.additional_job_roles_options.#{role}")
+  end
+
+  def public_job_roles
```
Comment (david-mears-dfe) [inline app/presenters/vacancy_presenter.rb:82]: I'm not sure what the meaning of `public` is here.

Comment (cesidio) [reply app/presenters/vacancy_presenter.rb:86]: as in public listing

Comment (david-mears-dfe) [reply app/presenters/vacancy_presenter.rb:86]: It seems to break the established convention of using 'show'?

File: spec/presenters/vacancy_presenter_spec.rb [hunk 2]
```diff
@@ -88,6 +88,22 @@
     end
   end
 
+  describe "#public_job_roles" do
+    subject { VacancyPresenter.new(vacancy) }
+
+    let(:vacancy) { build(:vacancy) }
+
+    it "returns a the primary job role" do
```
Comment (david-mears-dfe) [inline spec/presenters/vacancy_presenter_spec.rb:96]: "a the"

File: spec/presenters/vacancy_presenter_spec.rb [hunk 3]
```diff
@@ -88,6 +88,22 @@
...
+      expect(subject.public_job_roles).to include subject.show_primary_job_role
+    end
+
+    it "returns a the primary the additional job roles" do
```
Comment (david-mears-dfe) [inline spec/presenters/vacancy_presenter_spec.rb:100]: "a the primary the"

File: app/views/publishers/vacancies/edit_vacancy_sections/_edit_job_role.html.slim [hunk 4]
```diff
@@ -11,7 +11,8 @@
...
         html_attributes: { id: "primary_job_role" })
-      - component.slot(:row,
-        key: t("jobs.additional_job_roles"),
-        value: @vacancy.show_additional_job_roles.presence || t("jobs.not_defined"),
-        html_attributes: { id: "additional_job_roles" })
+      - unless @vacancy.primary_job_role == "sendco"
+        - component.slot(:row,
+          key: t("jobs.additional_job_roles"),
+          value: @vacancy.show_additional_job_roles.presence || t("jobs.not_defined"),
+          html_attributes: { id: "additional_job_roles" })
```
Comment (alexbowen) [inline app/views/publishers/vacancies/edit_vacancy_sections/_edit_job_role.html.slim:18]: doesnt this mean there will be duplicate html ids on the page?

Comment (alexbowen) [reply app/views/publishers/vacancies/edit_vacancy_sections/_edit_job_role.html.slim:18]: although i see you have actually introduced this

Comment (alexbowen) [reply app/views/publishers/vacancies/edit_vacancy_sections/_edit_job_role.html.slim:18]: maybe im wrong actually

**Review by alexbowen — APPROVED** [review]

**Review by csutter — APPROVED** [review]

**Review by david-mears-dfe — APPROVED** [review]

**Pull request merged**